templates/tag.twig

```diff
@@ -27,7 +27,7 @@
 	{% endif %}
 	<div class="tag_template">
 		<div class="container tag_header_container">
-			<h5 class="category">{{ category_name }}</h5>
+			<a class="category_name" href="/explore/{{ category_slug }}">{{ category_name }}</a>
```

Comment:
Since you know the slug, please retrieve the url the proper way https://developer.wordpress.org/reference/functions/get_page_by_path/